public-ui/src/app/views/survey-view/survey-view.component.ts

```diff
@@ -141,38 +165,28 @@ export class SurveyViewComponent implements OnInit, OnDestroy {
       },
       complete: () => { this.resultsComplete = true; }
     }));
-
-    // Filter when text value changes
-    this.subscriptions.push(
-      this.searchText.valueChanges
-        .debounceTime(400)
-        .distinctUntilChanged()
-        .subscribe((query) => { this.filterResults(); }));
-
-    // Set to loading as long as they are typing
-    this.subscriptions.push(this.searchText.valueChanges.subscribe(
-      (query) => localStorage.setItem('searchText', query)));
   }
 
   public setSurvey() {
     // Get the survey from local storage the user clicked on on a previous page
     const obj = localStorage.getItem('surveyModule');
     if (obj) {
-
       const survey = JSON.parse(obj);
       this.surveyConceptId = survey.conceptId;
       this.surveyPdfUrl = '/assets/surveys/' + survey.name.replace(' ', '_') + '.pdf';
+      this.getSurveyResults();
+    } else {
+      this.getThisSurvey();
```

Comment:
Is this needed anymore ? I am confused somehow. If the surveyModule is not present, someone would be redirected to getThisSurvey and then the survey object would be fetched and stored in local storage and the you come to set survey. Is the obj not present always ?